src/collocateElement.utility.js

```diff
@@ -536,32 +634,145 @@ const getRequestedCollocation = (modifiers = {}, defaultPosition = 'bottom', def
   }
 };
 
+const isTouchCloseButtonDisplayed = (touchCloseButton) => touchCloseButton?.style.display === 'block';
+
+const isTouchDropdownFullHeight = (element, rect) => {
+  rect = rect || element.getBoundingClientRect();
+  const viewport = getViewPort();
+  const maxAvailableViewportHeight = viewport.height - (maxTouchVerticalThresholdInPixels * 3);
+  return rect.height > maxAvailableViewportHeight;
+};
+
+const shouldDisplayCloseButton = (element, touchCloseButton) => !isTouchCloseButtonDisplayed(touchCloseButton) && isTouchDropdownFullHeight(element);
+
+const applyTouchElementContentMaxHeight = (element, elementContent) => {
+  if (!element || !elementContent) return;
+  const elementRect = element.getBoundingClientRect();
+  const contentExtraHeight = getContentExtraHeight(element, elementContent);
+  const dropdownHeight = elementRect.height;
+  const contentHeight = dropdownHeight - contentExtraHeight;
+  elementContent.style.maxHeight = `${contentHeight}px`;
+  elementContent.style.overflowY = 'auto';
+};
+
+const applyTouchElementCentering = (element, touchCloseButton) => {
+  if (!element) return;
+  const viewport = getViewPort();
+  const elementRect = element.getBoundingClientRect();
+  isTouchDropdownFullHeight(element, elementRect);
+
+  const top = window.scrollY + (viewport.height / 2);
+  const negativeMarginY = elementRect.height / 2;
+  element.style.top = `${top}px`;
+  element.style.marginTop = `-${negativeMarginY}px`;
+
+  const left = window.scrollX + (viewport.width / 2);
+  const negativeMarginX = elementRect.width / 2;
+  element.style.left = `${left}px`;
+  element.style.marginLeft = `-${negativeMarginX}px`;
+
+  if (touchCloseButton && isTouchDropdownFullHeight(element)) {
+    touchCloseButton.style.top = `${top}px`;
+    touchCloseButton.style.marginTop = `-${negativeMarginY + 15}px`;
+    element.style.marginTop = `-${negativeMarginY - 15}px`;
+    displayTouchCloseButton(touchCloseButton);
+  } else {
+    hideTouchCloseButton(touchCloseButton);
+  }
+};
+
+const applyTouchElementStyle = (element, touchCloseButton, computedElementDimensions) => {
+  const viewport = getViewPort();
+
+  const maxAvailableViewportHeight = viewport.height - (maxTouchVerticalThresholdInPixels * 2);
+  const maxAvailableHeight = Math.min(maxAvailableViewportHeight, computedElementDimensions.maxHeight);
+  element.style.maxHeight = `${maxAvailableHeight}px`;
+
+  const maxAvailableViewportWidth = viewport.width - (maxTouchHorizontalThresholdInPixels * 2);
+  const maxAvailableWidth = Math.min(maxAvailableViewportWidth, computedElementDimensions.maxWidth, maxTouchElementWidthInPixels);
+  element.style.maxWidth = `${maxAvailableWidth}px`;
+  element.style.width = `${maxAvailableWidth}px`;
+
+  applyTouchElementCentering(element, touchCloseButton);
+};
+
+const touchCollocateElemeAt = (element, elementContent, touchCloseButton, computedElementDimensions) => {
```

Comment:
```suggestion
const touchCollocateElementAt = (element, elementContent, touchCloseButton, computedElementDimensions) => {
```
I guess this is supposed to be `touchCollocateElementAt`, `touchCollocateElemAt` or  `touchCollocateElement`

Reply:
fixed